BASE/src/ServerTelemetryChannel/ServerTelemetryChannel.cs

```diff
@@ -357,6 +374,8 @@ public void Initialize(TelemetryConfiguration configuration)
                 throw new ArgumentNullException(nameof(configuration));
             }
 
+            ((ISupportCredentialEnvelope)this).CredentialEnvelope = configuration.CredentialEnvelope;
```

Comment:
can we just use `CredentialEnvelope=configuration.CredentialEnvelope;` instead of cast?

Reply:
see my comment above